### PR TITLE
Refactoring.

### PR DESCRIPTION
**Pull request by ygorshenin**
PTAL

### CODE REVIEW

File: openlr/openlr_simple_decoder.cpp [hunk 1]
```diff
@@ -250,84 +316,73 @@ class AStarRouter
...
       {
         continue;
       }
 
-      if (piU < kEps && stage + 1 < m_pivots.size())
+      if (piU < kEps)
       {
-        Vertex uu(u.m_junction, u.m_junction, ud /* stageStartDistance */, u.m_stage + 1);
+        Vertex v(u.m_junction, u.m_junction, ud /* stageStartDistance */, stage + 1);
+        bool const lastVertex = stage + 1 == m_pivots.size();
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp]: s/lastVertex/isLastVertex.

File: openlr/openlr_simple_decoder.cpp [hunk 2]
```diff
@@ -20,24 +21,32 @@
...
 #include "std/queue.hpp"
+#include "std/thread.hpp"
 #include "std/utility.hpp"
 
 using namespace std::rel_ops;
 using namespace routing;
 
 namespace  // A staff to get road data.
 {
+size_t constexpr kCLS = 64;
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp]: s/CLS/CacheLineSize

File: openlr/openlr_simple_decoder.cpp (hunk 1 above)
Comment (ygorshenin) [reply openlr/openlr_simple_decoder.cpp]: Done

File: openlr/openlr_simple_decoder.cpp (hunk 2 above)
Comment (ygorshenin) [reply openlr/openlr_simple_decoder.cpp]: Done

**Comment by ygorshenin** [comment]
PTAL

File: openlr/openlr_simple_decoder.cpp [hunk 3]
```diff
@@ -556,49 +615,90 @@ OpenLRSimpleDecoder::OpenLRSimpleDecoder(string const & dataFilename, Index cons
...
+  }
 
-    if (ref.m_points.size() == 2 && multipointsOnly)
-      continue;
+  if (segmentsToHandle != kHandleAllSegments && segmentsToHandle < segments.size())
+    segments.resize(segmentsToHandle);
+
+  vector<IRoadGraph::TEdgeVector> paths(segments.size());
+
+  size_t constexpr a = LCM(sizeof(LinearSegment), kCacheLineSize) / sizeof(LinearSegment);
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp]: Please add a comment what's going on here and that this is not the most important and valuable part of the program.

**Pull request merged**